mautic/mautic
# Fix html code with unicode content

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 4.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
This is PR to resolve https://github.com/mautic/mautic/issues/9491. More context what is issue https://github.com/mautic/mautic/issues/9491#issuecomment-748581544 
This issue bring this PR https://github.com/mautic/mautic/pull/9108 by @mabumusa1 - maybe you can help with it.
I've added unit test case, what we need resolve.




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create landing page with legacy builder
3. Add         ` <a href="https://m3.mautibox.com/3.x/media/images/testá.png">test with unicode</a>`
4. Apply in builder 3 times
5. Go to page 
6. Open source of page
7. You should see duplicate `<!DOCTYPE html>` tag in the header of page

![image](https://user-images.githubusercontent.com/462477/133920232-5b752877-4c64-452d-bc5b-1591deaa9f07.png)

8. Apply patch, create page from scratch and repeat all steps again
9.  `<!DOCTYPE html>` should be in the header just once




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9516"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
